test/test_va_api_fixture.cpp

```diff
@@ -332,8 +356,8 @@ void VAAPIFixture::querySurfaceAttributes(SurfaceAttributes& attribs) const
 void VAAPIFixture::getMinMaxSurfaceResolution(
     Resolution& minRes, Resolution& maxRes) const
 {
-    const Resolution::DataType maxVal =
-        std::numeric_limits<Resolution::DataType>::max();
+    auto maxFn = std::numeric_limits<Resolution::DataType>::max;
+    const Resolution::DataType maxVal = maxFn();
```

Comment:
why this change was made?

Reply:
There's a `#define max(a,b)            (((a) > (b)) ? (a) : (b))` definition in `minwindef.h` which gets pulled when building on Windows and conflicts with that syntax, producing a build fail otherwise. I will add a comment to it.

Reply:
I know about this definition, but I don't understand how your change helps to avoid it? Basically you still have `max` in the code and it should get substituted producing something which should not compile, i.e.: 
```
auto maxFn = std::numeric_limits<Resolution::DataType>::(((a) > (b)) ? (a) : (b));
```

In other projects we were resolving this problem by adding `NOMINMAX` which from my perspective is a better way.